src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -76,8 +72,13 @@ public boolean equals(Object o) {
         }
 
         DeleteCommand other = (DeleteCommand) o;
-        return toDeleteName.equals(other.toDeleteName)
-                || toDeleteIndex.equals(other.toDeleteIndex); // state check
+        if (toDeleteName != null && other.toDeleteName != null) {
+            return toDeleteName.equals(other.toDeleteName);
+        } else if (toDeleteIndex != null && other.toDeleteIndex != null) {
+            return toDeleteIndex.equals(other.toDeleteIndex);
+        } else {
+            return false;
+        }
```

Comment:
Think this works for now, can consider using optionals during the refactor phase (1.4)!